mscore/preferenceslistwidget.cpp

```diff
@@ -215,103 +327,496 @@ DoublePreferenceItem::DoublePreferenceItem(QString name)
+            applyFunction();
+      }
+
+QWidget* StringPreferenceItem::editor() const
+      {
+      if (_editor)
+            return _editor;
+      else if (_editor2)
+            return _editor2;
+      else if (_editor3)
+            return _editor3;
+      else
+            return nullptr;
+      }
+
+bool StringPreferenceItem::isModified() const // I have to add modified
+      {
+      if (_editor)
+            return _initialValue != _editor->text();
+      else if (_editor2)
+            return _initialValue != _editor2->currentFont().family();
+      else if (_editor3)
+            return _initialValue != _editor3->currentText(); // this should be currentData.toString() but this causes a crash INVESTIGATE!
+      else if (_editor4) { // does this work? is this needed?
+            QString currentValue = _initialValue;
+            applyFunction();
+            QString newValue = _initialValue;
+            preferences.setPreference(name(), currentValue);
+            return _initialValue != newValue;
+            }
+      else
+            Q_ASSERT(false);
       }
 
-bool StringPreferenceItem::isModified() const
+//---------------------------------------------------------
+//   CustomPreferenceItem
+//---------------------------------------------------------
+
+CustomPreferenceItem::CustomPreferenceItem(QString name, QRadioButton* editor, std::function<void ()> applyFunc, std::function<void ()> updateFunc)
+      : PreferenceItem(name),
+        _editor(editor)
+      {
+      Q_ASSERT(applyFunc.operator bool()); // if an apply and an update function are not provided this cannot work
+      applyFunction = applyFunc;
+      Q_ASSERT(updateFunc.operator bool());
+      updateFunction = updateFunc;
+      updateFunction(); // update on creation is the same as assigning an initival value
+      connect(_editor, &QRadioButton::toggled, this, &PreferenceItem::editorValueModified);
+      }
+
+CustomPreferenceItem::CustomPreferenceItem(QString name, QComboBox* editor, std::function<void ()> applyFunc, std::function<void ()> updateFunc)
+      : PreferenceItem(name),
+        _editor2(editor)
+      {
+      Q_ASSERT(applyFunc.operator bool()); // if an apply and an update function are not provided this cannot work
+      applyFunction = applyFunc;
+      Q_ASSERT(updateFunc.operator bool());
+      updateFunction = updateFunc;
+      updateFunction(); // update on creation is the same as assigning an initival value
+      connect(_editor2, QOverload<int>::of(&QComboBox::currentIndexChanged), this, &PreferenceItem::editorValueModified);
+      }
+
+void CustomPreferenceItem::save()
       {
-      return _initialValue != _editor->text();
+      applyFunction();
       }
 
+void CustomPreferenceItem::update()
+      {
+      updateFunction();
+      }
+
+void CustomPreferenceItem::setDefaultValue()
+      {
+      Q_ASSERT(false);
+      }
 
+QWidget* CustomPreferenceItem::editor() const
+      {
+      Q_ASSERT(false);
```

Comment:
it may not be needed, but you could do the following here:
```
QRadioButton* ed = static_cast<QRadioButton*>(_editor);
if (ed)
    return ed;

QComboBox* ed2 = static_cast<QComboBox*>(_editor);
if (ed2)
    return ed2;

// fallback
return nullptr;
```